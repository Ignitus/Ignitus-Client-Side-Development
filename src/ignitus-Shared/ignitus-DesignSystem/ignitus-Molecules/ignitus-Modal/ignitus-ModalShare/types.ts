export type ModalShareProps = {
  title: string;
  Share: string;
  Link: string;
  LinkedinColor: string;
  TwitterColor: string;
  RedditColor: string;
  FacebookColor: string;
  White: string;
  GreyOne: string;
};
