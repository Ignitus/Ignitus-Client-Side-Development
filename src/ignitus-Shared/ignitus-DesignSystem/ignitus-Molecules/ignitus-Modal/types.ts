export type Props = {
  secondary?: boolean;
};

export type ModalProps = {
  primaryText: string;
  secondaryText: string;
  title: string;
  buttonText: string;
};
