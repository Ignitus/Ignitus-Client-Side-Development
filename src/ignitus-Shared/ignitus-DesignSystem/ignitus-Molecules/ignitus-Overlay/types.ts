export type OverlayProps = {
  primaryText: string;
  secondaryText: string;
};
