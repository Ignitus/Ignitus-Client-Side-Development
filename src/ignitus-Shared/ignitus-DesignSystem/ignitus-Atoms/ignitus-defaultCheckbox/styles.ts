import styled from '@emotion/styled';
import { Paragraph } from '../typography';
import { LabelType } from './types';

export const CheckBox = styled.input`
  height: 1rem;
  width: 1rem;
  outline: none;
`;
export const Label = styled.label`
  margin-left: 0.4rem;
`;

export const StyledParagraph = styled(Paragraph)<LabelType>`
  font-size: ${props => props.size};
`;

export const Container = styled.div`
  display: inline-flex;
  align-items: center;
`;
