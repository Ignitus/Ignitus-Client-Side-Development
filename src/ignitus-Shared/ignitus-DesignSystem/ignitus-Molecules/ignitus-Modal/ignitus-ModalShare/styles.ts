import styled from '@emotion/styled';
import { Paragraph as P, Normal, MD, White, boxShadowColor } from '../../../..';
import { Icon } from '../../../../ignitus-Utilities/Components/icon';

import { flexibleColDiv, flexibleRowDiv } from '../../../shared';

export const ModalShare = styled.div`
  background-color: ${White};
  background-size: contain;
  border-radius: 1rem;
  display: inline-flex;
  flex-direction: column;
  box-shadow: 0 2px 4px 0 ${boxShadowColor};
  width: 35rem;
`;

export const TopSection = styled(flexibleRowDiv)`
  padding: 1.5rem;
  justify-content: space-between;
`;

export const BottomSection = styled(flexibleColDiv)`
  padding: 0.4rem 0.8rem 1.5rem 1.5rem;
  align-items: flex-start;
`;

export const IconsContainer = styled(flexibleRowDiv)`
  border-radius: 1rem;
  display: inline-flex;
`;

export const IconContainer = styled.button`
  border: none;
  cursor: pointer;
  overflow: hidden;
  margin: 0 0.2rem;
  height: 2.8rem;
  width: 2.8rem;
  border-radius: 50%;
  background-color: ${props => props.color};
  outline-color: transparent;
`;

export const Paragraph = styled(P)`
  font-weight: ${Normal};
  font-size: ${MD};
  margin: 2rem 0.6rem 0.6rem 0.6rem;
`;

export const StyledIcon = styled(Icon)`
  width: 1.4rem;
  fill: ${props => props.color};
  transition: 0.3s;
  flex: 1;
  margin-top: 0.2rem;
`;
