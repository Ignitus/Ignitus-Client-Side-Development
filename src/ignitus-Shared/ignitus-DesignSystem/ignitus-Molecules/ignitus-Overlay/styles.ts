import styled from '@emotion/styled';
import { flexibleColDiv } from '../../shared';
import { Icon } from '../../../ignitus-Utilities/Components/icon';
import { Heading1 } from '../../..';

export const Overlay = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgba(0, 0, 102, 0.89);
  backdrop-filter: blur(10px);
  width: 100%;
  height: 100%;
`;

export const Logo = styled(Icon)`
  margin-bottom: 4rem;
  height: 7.5rem;
  path {
    fill: white;
  }
`;

export const Heading = styled(Heading1)`
  margin-bottom: 1.5rem;
`;

export const MessageSection = flexibleColDiv;
