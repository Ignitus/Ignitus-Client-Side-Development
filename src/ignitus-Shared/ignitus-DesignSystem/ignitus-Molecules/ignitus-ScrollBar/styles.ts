import styled from '@emotion/styled';

export const Scroll = styled.div`
  margin: 30px;
  float: left;
  height: 300px;
  width: 65px;
  background: #f5f5f5;
  overflow-y: scroll;
  margin-bottom: 25px;

  ::-webkit-scrollbar {
    width: 12px;
    background: #f5f5f5;
  }

  ::-webkit-scrollbar-track {
    box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.3);
    border-radius: 10px;
    background-color: #f5f5f5;
  }

  ::-webkit-scrollbar-thumb {
    border-radius: 10px;
    background-color: #f90;
  }
`;

export const Overflow = styled.div`
  min-height: 450px;
`;
