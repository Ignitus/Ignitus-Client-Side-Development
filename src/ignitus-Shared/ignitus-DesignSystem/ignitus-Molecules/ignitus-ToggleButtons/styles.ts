import styled from '@emotion/styled';

import { Icon } from '../../../ignitus-Utilities/Components/icon';
import * as B from '../../ignitus-Atoms/buttons';

export const ButtonLeft = styled(B.Button)`
  font-weight: 600;
  margin: 0;
  border: 0;
  float: left;
  border-radius: 20px 0 0 20px;
`;

export const ButtonRight = styled(B.Button)`
  font-weight: 600;
  margin: 0;
  border: 0;
  float: left;
  border-radius: 0 20px 20px 0;
`;

export const ToggleButtonContainer = styled.span`
  cursor: pointer;
`;

export const StyledIcon = styled(Icon)`
  width: 1.4rem;
  fill: ${props => props.color};
  transition: 0.3s;
  flex: 1;
  margin: 0.03rem;
`;
