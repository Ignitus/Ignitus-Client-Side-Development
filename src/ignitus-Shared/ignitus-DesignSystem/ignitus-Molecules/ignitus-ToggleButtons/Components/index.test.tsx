import React from 'react';
import { mount, render } from 'enzyme';
import emotionSerializer, { matchers } from 'jest-emotion';

import { ToggleButton, ToggleIconButton } from './index';
import { ToggleButtonProps, ToggleIconButtonProps } from '../types';
import { AppIcon } from '../../../../types/iconsTypes/iconEnums';
import { StyledIcon } from '../styles';
import { IgnitusBlue } from '../../../..';

const propsButton: ToggleButtonProps = {
  leftLabel: 'left',
  rightLabel: 'right',
};

const propsButtonIcon: ToggleIconButtonProps = {
  leftIcon: AppIcon.GridIcon,
  rightIcon: AppIcon.BarsIcon,
};

const propsStyledIcon = {
  name: AppIcon.GridIcon,
  color: `${IgnitusBlue}`,
};

const colorCheck = () => {
  const Icon = mount(<StyledIcon {...propsStyledIcon} />);
  expect(Icon).toHaveStyleRule('fill', propsStyledIcon.color);
};

expect.addSnapshotSerializer(emotionSerializer);
expect.extend(matchers);

it('should rendered', () => {
  const tree = render(<ToggleButton {...propsButton} />);
  expect(tree).toMatchSnapshot();
});

describe('<ToggleButton />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<ToggleButton {...propsButton} />);
  });

  it('should render ToggleButton with correct labels', () => {
    expect(
      wrapper
        .find('button')
        .at(0)
        .text()
        .trim(),
    ).toBe(propsButton.leftLabel);
    expect(
      wrapper
        .find('button')
        .at(1)
        .text()
        .trim(),
    ).toBe(propsButton.rightLabel);
  });
});

it('should rendered', () => {
  const tree = render(<ToggleIconButton {...propsButtonIcon} />);
  expect(tree).toMatchSnapshot();
});

describe('<ToggleIconButton />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<ToggleIconButton {...propsButtonIcon} />);
  });

  it('should render ToggleButtonIcon with correct icons', () => {
    expect(
      wrapper
        .find(StyledIcon)
        .at(0)
        .prop('name'),
    ).toBe(propsButtonIcon.leftIcon);
    expect(
      wrapper
        .find(StyledIcon)
        .at(1)
        .prop('name'),
    ).toBe(propsButtonIcon.rightIcon);
  });

  it('should render StyledIcon with color passed in prop', colorCheck);
});
