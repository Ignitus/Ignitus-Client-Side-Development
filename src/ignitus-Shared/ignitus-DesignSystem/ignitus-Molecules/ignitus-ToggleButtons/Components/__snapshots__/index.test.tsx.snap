// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendered 1`] = `
.emotion-2 {
  cursor: pointer;
}

.emotion-0 {
  border-radius: 5px;
  border: none;
  cursor: pointer;
  outline: none;
  padding: 16px 32px;
  text-align: center;
  line-height: 1rem;
  margin: 0.5rem;
  outline: none;
  font-size: 18px;
  font-weight: 400;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  outline: none;
  color: #ffffff;
  background-color: #000066;
  padding: 8px 20px;
  font-size: 16px;
  border-radius: 3px;
  border: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  margin: 0;
  border: 0;
  float: left;
  border-radius: 20px 0 0 20px;
}

.emotion-0:hover,
.emotion-0:focus {
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.05);
}

.emotion-0 a {
  color: #ffffff;
}

.emotion-0:hover,
.emotion-0:focus {
  background-color: #F9BF64;
  color: #000066;
  outline: none;
}

.emotion-0:hover a,
.emotion-0:focus a {
  color: #000066;
}

.emotion-0:hover svg,
.emotion-0:focus svg {
  fill: #000066;
}

.emotion-1 {
  border-radius: 5px;
  border: none;
  cursor: pointer;
  outline: none;
  padding: 16px 32px;
  text-align: center;
  line-height: 1rem;
  margin: 0.5rem;
  outline: none;
  font-size: 18px;
  font-weight: 400;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  outline: none;
  color: #000066;
  background-color: #E0E0E0;
  padding: 8px 20px;
  font-size: 16px;
  border-radius: 3px;
  border: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  margin: 0;
  border: 0;
  float: left;
  border-radius: 0 20px 20px 0;
}

.emotion-1:hover,
.emotion-1:focus {
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.05);
}

.emotion-1 a {
  color: #000066;
}

.emotion-1:hover,
.emotion-1:focus {
  background-color: #C4C4C4;
  color: #000066;
  outline: none;
}

.emotion-1:hover a,
.emotion-1:focus a {
  color: #000066;
}

.emotion-1:hover svg,
.emotion-1:focus svg {
  fill: #000066;
}

<span
  class="emotion-2"
>
  <button
    class="emotion-0"
  >
     left 
  </button>
  <button
    class="emotion-1"
  >
     right
  </button>
</span>
`;

exports[`should rendered 2`] = `
.emotion-4 {
  cursor: pointer;
}

.emotion-1 {
  border-radius: 5px;
  border: none;
  cursor: pointer;
  outline: none;
  padding: 16px 32px;
  text-align: center;
  line-height: 1rem;
  margin: 0.5rem;
  outline: none;
  font-size: 18px;
  font-weight: 400;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  outline: none;
  color: #000066;
  background-color: #E0E0E0;
  padding: 8px 20px;
  font-size: 16px;
  border-radius: 3px;
  border: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  margin: 0;
  border: 0;
  float: left;
  border-radius: 20px 0 0 20px;
}

.emotion-1:hover,
.emotion-1:focus {
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.05);
}

.emotion-1 a {
  color: #000066;
}

.emotion-1:hover,
.emotion-1:focus {
  background-color: #C4C4C4;
  color: #000066;
  outline: none;
}

.emotion-1:hover a,
.emotion-1:focus a {
  color: #000066;
}

.emotion-1:hover svg,
.emotion-1:focus svg {
  fill: #000066;
}

.emotion-0 {
  width: 1.4rem;
  fill: #000066;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 0.03rem;
}

.emotion-3 {
  border-radius: 5px;
  border: none;
  cursor: pointer;
  outline: none;
  padding: 16px 32px;
  text-align: center;
  line-height: 1rem;
  margin: 0.5rem;
  outline: none;
  font-size: 18px;
  font-weight: 400;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  outline: none;
  color: #000066;
  background-color: #ffffff;
  padding: 8px 20px;
  font-size: 16px;
  border-radius: 3px;
  border: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  margin: 0;
  border: 0;
  float: left;
  border-radius: 0 20px 20px 0;
}

.emotion-3:hover,
.emotion-3:focus {
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.05);
}

.emotion-3 a {
  color: #000066;
}

.emotion-3:hover,
.emotion-3:focus {
  background-color: none;
  color: none;
  outline: none;
}

.emotion-3:hover a,
.emotion-3:focus a {
  color: #000066;
}

.emotion-3:hover svg,
.emotion-3:focus svg {
  fill: #000066;
}

<span
  class="emotion-4"
>
  <button
    class="emotion-1"
  >
    <svg
      aria-label="Grid Icon"
      class="emotion-0"
      color="#000066"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Grid Icon
      </title>
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M4 11h5V5H4v6zm0 7h5v-6H4v6zm6 0h5v-6h-5v6zm6 0h5v-6h-5v6zm-6-7h5V5h-5v6zm6-6v6h5V5h-5z"
      />
    </svg>
  </button>
  <button
    class="emotion-3"
  >
    <svg
      aria-label="Bars Icon"
      class="emotion-0"
      color="#000066"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Bars Icon
      </title>
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
</span>
`;
